### PR TITLE
[ARM] Change the type of CC and VCC code in `splitMnemonic`.

### PR DESCRIPTION
**Pull request by AlfieRichardsArm**
This changes the type of `PredicationCode` and `VPTPredicationCode` from `unsigned` to `ARMCC::CondCodes` and  `ARMVCC::VPTCodes` resp' for clarity and correctness.

### CODE REVIEW

**Comment by AlfieRichardsArm** [comment]
Note the failing test is not relevant, seems to be a timeout or similar

**Comment by AlfieRichardsArm** [comment]
I will wait a couple days for external code review

File: llvm/lib/Target/ARM/AsmParser/ARMAsmParser.cpp [hunk 1]
```diff
@@ -6340,7 +6340,7 @@ StringRef ARMAsmParser::splitMnemonic(StringRef Mnemonic,
     unsigned CC = ARMCondCodeFromString(Mnemonic.substr(Mnemonic.size()-2));
     if (CC != ~0U) {
       Mnemonic = Mnemonic.slice(0, Mnemonic.size() - 2);
-      PredicationCode = CC;
+      PredicationCode = (ARMCC::CondCodes)CC;
```
Comment (s-barannikov) [inline llvm/lib/Target/ARM/AsmParser/ARMAsmParser.cpp:6343]: https://llvm.org/docs/CodingStandards.html#prefer-c-style-casts

**Review by s-barannikov — APPROVED** [review]
Straightforward change, LGTM modulo style nit

File: llvm/lib/Target/ARM/AsmParser/ARMAsmParser.cpp (hunk 1 above)
Comment (AlfieRichardsArm) [reply llvm/lib/Target/ARM/AsmParser/ARMAsmParser.cpp:6343]: The anchor link really changes the meaning of that link (:
I'll get a quick fix in

**Comment by AlfieRichardsArm** [comment]
Failing test continues to be a seemingly unrelated failure regarding MLIR

**Comment by AlfieRichardsArm** [comment]
After a fixing me pushing to the wrong branch this is back to what it should be.

**Comment by AlfieRichardsArm** [comment]
Test failure is unrelated MLIR test. Merging this now.

**Pull request merged**